src/style.css

```diff
@@ -0,0 +1,32 @@
+* {
+  margin: initial;
+  box-sizing: border-box;
+}
+
+.stars {
+  display: flex;
+}
+
+.stars__star {
+  height: 15px;
+  width: 15px;
+  background-repeat: no-repeat;
+  background-position: center;
+  margin-right: 4px;
+}
+
+.stars--0 > .stars__star,
+.stars--1 .stars__star:nth-child(n + 2),
+.stars--2 .stars__star:nth-child(n + 3),
+.stars--3 .stars__star:nth-child(n + 4),
+.stars--4 .stars__star:nth-child(n + 5) {
+  background-image: url(images/star.svg);
+}
```

Comment:
Let's simplify your code a little bit.
You can set this as a default background image for `.stars__star` for the same result.